tezos/context/src/chunks/vec.rs

```diff
@@ -0,0 +1,378 @@
+
+            let (slice, rest) = if remaining_slice.len() > space_in_chunk {
+                remaining_slice.split_at(space_in_chunk)
+            } else {
+                (remaining_slice, &[][..])
+            };
+
+            remaining_slice = rest;
+            last_chunk.extend_from_slice(slice);
+        }
+
+        self.nelems += slice_length;
+        (start, slice_length)
+    }
+
+    /// Appends `other` in the last chunk.
+    ///
+    /// Return the index of the slice in the chunks, and its length
+    pub fn append(&mut self, other: &mut Vec<T>) -> (usize, usize) {
+        let (start, length) = self.extend_from_slice(other);
+        other.truncate(0);
+
+        (start, length)
+    }
+
+    pub fn extend_from(&mut self, other: &Self) {
+        let our_length = self.list_of_chunks.len();
+        let other_length = other.list_of_chunks.len();
+
+        if our_length != other_length {
+            assert!(our_length < other_length);
+            self.list_of_chunks
+                .resize_with(other_length, Default::default);
+        }
+
+        let our_length = our_length.saturating_sub(1);
+        let mut nelems = 0;
+
+        for (ours, other) in self.list_of_chunks[our_length..]
+            .iter_mut()
+            .zip(&other.list_of_chunks[our_length..])
+        {
+            let ours_length = ours.len();
+            if ours_length < other.len() {
+                nelems += other.len() - ours_length;
+                ours.extend_from_slice(&other[ours_length..]);
+            }
+        }
+
+        self.nelems += nelems;
+        assert_eq!(self.nelems, other.nelems);
+    }
+
+    pub fn get_slice(&self, Range { start, end }: Range<usize>) -> Option<Cow<[T]>> {
+        let slice_length = end - start;
+        let (list_index, chunk_index) = self.get_indexes_at(start);
+
+        let chunk = self.list_of_chunks.get(list_index)?;
+
+        if chunk_index + slice_length <= self.chunk_capacity {
+            chunk
+                .get(chunk_index..chunk_index + slice_length)
+                .map(Cow::Borrowed)
+        } else {
+            let mut slice = Vec::with_capacity(slice_length);
+            let mut iter_chunk = self.list_of_chunks.get(list_index..)?.iter();
+            let mut start_in_chunk = chunk_index;
+            let mut length = slice_length;
+
+            while length > 0 {
+                let chunk = iter_chunk.next()?;
+                let end_in_chunk = (start_in_chunk + length).min(self.chunk_capacity);
+
+                let part_slice = chunk.get(start_in_chunk..end_in_chunk)?;
+                slice.extend_from_slice(part_slice);
+
+                length -= end_in_chunk - start_in_chunk;
+                start_in_chunk = 0;
+            }
+
+            debug_assert_eq!(slice.len(), slice_length);
+
+            Some(Cow::Owned(slice))
+        }
+    }
+}
+
+impl<T> ChunkedVec<T> {
+    /// Returns a new `ChunkedSlice<T>` without allocating
```

Comment:
`ChunkedSlice` is mentioned a few times in the docs and examples but it doesn't exist anymore.